packages/plugins/plugin-builder-flowtype/src/transformer/index.ts

```diff
@@ -0,0 +1,276 @@
+                  t.arrayExpression(node.types.map(transformLiteral)),
+                ]);
+              case 'ObjectTypeAnnotation':
+                return createRuntimeFunctionCall('TypeLiteral', [
+                  t.functionExpression(
+                    null,
+                    [t.identifier('typeLiteral')],
+                    t.blockStatement([
+                      // { key: value }
+                      ...node.properties.map((property) => {
+                        if (property.type === 'ObjectTypeProperty') {
+                          return t.expressionStatement(
+                            t.callExpression(
+                              t.memberExpression(
+                                t.identifier('typeLiteral'),
+                                t.identifier('property'),
+                              ),
+                              [
+                                property.key.type === 'Identifier'
+                                  ? t.stringLiteral(property.key.name)
+                                  : property.key,
+                                transformLiteral(property.value),
+                                t.booleanLiteral(!!property.optional),
+                                // TODO:
+                                t.arrayExpression(),
+                              ],
+                            ),
+                          );
+                        } else {
+                          // TODO: spread property
+                          return t.emptyStatement();
+                        }
+                      }),
+                      // { [key: type]: value }
+                      ...(node.indexers
+                        ? node.indexers.map((indexer) => {
+                            return t.expressionStatement(
+                              t.callExpression(
+                                t.memberExpression(
+                                  t.identifier('typeLiteral'),
+                                  t.identifier('computedProperty'),
+                                ),
+                                [
+                                  t.stringLiteral(
+                                    indexer.id ? indexer.id.name : 'key',
+                                  ),
+                                  transformLiteral(indexer.key),
+                                  transformLiteral(indexer.value),
+                                  t.numericLiteral(0), // ComputedPropertyOperator.INDEX_SIGNATURE
+                                  t.booleanLiteral(false),
+                                  t.arrayExpression(),
+                                ],
+                              ),
+                            );
+                          })
+                        : []),
+                    ]),
+                  ),
+                ]);
+              case 'NullableTypeAnnotation':
+                // ?T -> runtime.OptionalType(T)
```

Comment:
`?T` in FlowType is not equivalent to `OptionalType` in TS, where `OptionalType` is only used in `TupleType`s. (i.e: `[string, number, number?]`. In other places, `?` is an enum (`QuestionToken`) in object properties.

So we might want to create a `NullableType` in `runtime`, and we should as we aim to have first-class support for Flow.